cli-commands/verifier/verifier.js

```diff
@@ -0,0 +1,169 @@
+		contractWrapper.interface.deployFunction.inputs.forEach(element => {
+			types.push(element.type);
+		});
+		let encodedConstructorArgs = (await encoder.encode(types, deploymentArguments)).substr(2);
+		return encodedConstructorArgs;
+	}
+
+	_buildLibrariesArguments(libraries) {
+		if (!libraries || Object.keys(libraries).length === 0) {
+			return
+		}
+		let arrayLib = Object.entries(libraries).map(([key, value]) => ({ name: key, address: value }));
+		return arrayLib
+	}
+
+	_buildSolcVersionCompiler(solcVersion) {
+		const version = `v${solcVersion.replace(/.Emscripten.clang/g, '')}`;
+		return version;
+	}
+
+	async _buildApiUrl(provider) {
+		const { name } = await provider.getNetwork();
+		this.name = name;
+
+		if ((/^(ropsten|rinkeby|kovan|goerli)$/.test(this.name))) {
+			return `https://api-${this.name}.etherscan.io/api`;
+		}
+		return 'https://api.etherscan.io/api';
+	}
+
+	_constructRequestData() {
+		let data = {
+			apikey: this.apiKey,
+			module: 'contract', // DO NOT CHANGE
+			action: 'verifysourcecode', // DO NOT CHANGE
+			contractaddress: this.contractAddress,
+			sourceCode: this.flattenCode,
+			contractname: this.contractName,
+			compilerversion: this.solcVersionCompiler,
+			optimizationUsed: this.optimization,
+			runs: this.runs,
+			constructorArguements: this.constructorArguments
+		}
+		if (this.libraries) {
+			for (let i = 0; i < this.libraries.length; i++) {
+				data[`libraryname${i + 1}`] = this.libraries[i].name;
+				data[`libraryaddress${i + 1}`] = this.libraries[i].address;
+			}
+		}
+		let stringData = querystring.stringify(data);
+		return stringData
+	}
+
+	async _sendVerificationRequest(stringData, defaultOverrides) {
+
+		const ms = defaultOverrides.waitInterval || DEFAULT_SEND_REQUEST_TIMEOUT;
+		let count = 0;
+		const self = this;
+		async function sendRequest(ms) {
+			await self.timeout(ms)
+			const response = await axios.post(self.apiUrl, stringData);
+			if (count < 10) {
+				if (response.data.message == 'OK') {
+					logger.log(`Request successfully sent, your GUID is ${colors.colorParams(response.data.result)}`);
+					return response.data
+				} else {
+					console.log('Processing verification. Please wait. It might take a few minutes');
+					count++;
+					return await sendRequest(ms);
+				}
+			} else {
+				throw new Error('Contract Verification Timeout!!! Please try again');
+			}
+		}
+
+		return sendRequest(ms);
+
+	}
+
+	async _checkVerificationStatus(response, defaultOverrides) {
+		let params = {
+			guid: response.result,
+			module: "contract", // DO NOT CHANGE
+			action: "checkverifystatus" // DO NOT CHANGE
+		};
+		const ms = defaultOverrides.waitInterval || DEFAULT_CHECK_STATUS_TIMEOUT;
+		let count = 0;
+		const self = this;
+		async function checkGuid(ms) {
+			await self.timeout(ms);
+			const response = await axios.get(self.apiUrl, { params });
+			if (count < 10) {
+				if (response.data.result == 'Pending in queue') {
+
+					console.log('Pending...');
```

Comment:
I think we were using a logger object instead of console.log, in order to be able to disable the logs. Can you check and change it if I am right